prow/flagutil/github.go

```diff
@@ -47,25 +47,62 @@ type GitHubOptions struct {
+		graphqlEndpoint: github.DefaultGraphQLEndpoint,
+	}
+}
+
+type CustomGitHubDefaulter func(options *GitHubOptions)
+
+func GitHubThrottlerDefaults(hourlyTokens, allowedBursts int) CustomGitHubDefaulter {
+	return func(o *GitHubOptions) {
+		o.ThrottleHourlyTokens = hourlyTokens
+		o.ThrottleAllowBurst = allowedBursts
+	}
+}
+
 // AddFlags injects GitHub options into the given FlagSet.
 func (o *GitHubOptions) AddFlags(fs *flag.FlagSet) {
-	fs.StringVar(&o.Host, "github-host", github.DefaultHost, "GitHub's default host (may differ for enterprise)")
-	o.endpoint = NewStrings(github.DefaultAPIEndpoint)
+	defaults := githubDefaults()
+	o.addFlags(fs, defaults)
+}
+
+// AddFlagsWithCustomDefaults injects GitHub options into the given FlagSet,
+func (o *GitHubOptions) AddFlagsWithCustomDefaults(fs *flag.FlagSet, defaulters ...CustomGitHubDefaulter) {
+	defaults := githubDefaults()
+	for _, defaulter := range defaulters {
+		defaulter(&defaults)
+	}
+	o.addFlags(fs, defaults)
+}
+
+// AddFlagsWithCustomDefaults injects GitHub options into the given FlagSet
+func (o *GitHubOptions) addFlags(fs *flag.FlagSet, defaults GitHubOptions) {
+	fs.StringVar(&o.Host, "github-host", defaults.Host, "GitHub's default host (may differ for enterprise)")
+	o.endpoint = NewStrings(defaults.endpoint.Strings()...)
 	fs.Var(&o.endpoint, "github-endpoint", "GitHub's API endpoint (may differ for enterprise).")
-	fs.StringVar(&o.graphqlEndpoint, "github-graphql-endpoint", github.DefaultGraphQLEndpoint, "GitHub GraphQL API endpoint (may differ for enterprise).")
-	fs.StringVar(&o.TokenPath, "github-token-path", "", "Path to the file containing the GitHub OAuth secret.")
-	fs.StringVar(&o.AppID, "github-app-id", "", "ID of the GitHub app. If set, requires --github-app-private-key-path to be set and --github-token-path to be unset.")
-	fs.StringVar(&o.AppPrivateKeyPath, "github-app-private-key-path", "", "Path to the private key of the github app. If set, requires --github-app-id to bet set and --github-token-path to be unset")
+	fs.StringVar(&o.graphqlEndpoint, "github-graphql-endpoint", defaults.graphqlEndpoint, "GitHub GraphQL API endpoint (may differ for enterprise).")
+	fs.StringVar(&o.TokenPath, "github-token-path", defaults.TokenPath, "Path to the file containing the GitHub OAuth secret.")
+	fs.StringVar(&o.AppID, "github-app-id", defaults.AppID, "ID of the GitHub app. If set, requires --github-app-private-key-path to be set and --github-token-path to be unset.")
+	fs.StringVar(&o.AppPrivateKeyPath, "github-app-private-key-path", defaults.AppPrivateKeyPath, "Path to the private key of the github app. If set, requires --github-app-id to bet set and --github-token-path to be unset")
+	fs.IntVar(&o.ThrottleHourlyTokens, "github-hourly-tokens", defaults.ThrottleHourlyTokens, "If set to a value larger than zero, enable client-side throttling to limit hourly token consumption. If set, --github-allowed-burst must be positive too.")
```

Comment:
I was kind of expecting the flag adding to be conditional

Reply:
Why? I can make it so, but is there a good reason to make them so?

Reply:
Yes, something like tide will have multiple flags for this and one of it just doesn't do anything which is confusing at best

Reply:
Maybe I misunderstood, but I thought you suggested the https://github.com/kubernetes/test-infra/pull/21891#discussion_r617678748 to allow opting out.

Reply:
I wrote:
> Can we maybe just add an option to not add the new throttler flags if requested and use that?

Adding a bool option to the flagutil to not add these two flags if explicitly requested is what I meant by that